Refine README and other docs

Gnnng opened the pull request:
## Description
<!-- What is the overall goals of your pull request? -->
<!-- What is the context of your pull request? -->
<!-- What are the related issues and pull requests? -->

This PR includes updates in README and other docs. Discussions and contributions are welcome.

## Review Suggestion
<!-- How should the reviewers get started on reviewing your pull request-->
<!-- How can the reviewers verify the pull request is working as expected -->

N/A

## Status

### Implementation
<!-- Add more TODOs before "ready for review", if any  -->
- [x] GPU mining doc
- [x] **ready for review**

### Integration Test (Core Team)
<!-- This is for core team only, ignore this if you are a community contributor -->
- [ ] local brain test (please proofread :smiley:)

bb111189 commented:
@ansnunez might want to chip in to refine the readme on the pow part to reflect the current design (merged pow)

Gnnng commented on README.md:
```diff
@@ -51,63 +59,62 @@ To run Zilliqa, we recommend the following minimum system requirements:
+The OpenCL support is build into the Zilliqa by default. If you have a graphics card with AMD GPU, you can use it for PoW already.
+
+If you wish to use CUDA supported GPU for PoW, please download and install CUDA package from [NVIDIA offical webpage](https://developer.nvidia.com/cuda-downloads). You may need to reboot your PC for the installation take effect. After which, please use the following build option to build Zilliqa with CUDA support.
 `./build.sh cuda`
```
@ShengguangXiao can you sqaush your commits into one?

Also, let's put `./build.sh cuda` in a code block like this

```
./build.sh cuda
```

bb111189 commented:
I think we should also put transactions sharding into available feature

Gnnng commented:
Updated according to @bb111189's suggestions and added our log to README.md. 

Note that the image `src` href points to the image on master branch after this PR is merged, so now it shows blank in the markdown preview. If you want to see the overall look, use this [edit link](https://github.com/Zilliqa/Zilliqa/edit/fix/readme/README.md?pr=%2FZilliqa%2FZilliqa%2Fpull%2F508) and replace `master` with `fix/readme` in the image `src`.

```
<img src="https://github.com/Zilliqa/Zilliqa/blob/fix/readme/img/zilliqa-logo-color.png" width="200" height="200">
```
